src/GraphQL/Federation/Resolvers/EntityResolver.cs

```diff
@@ -0,0 +1,191 @@
+using System.Collections;
+using GraphQL.DataLoader;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using static GraphQL.Federation.FederationHelper;
+
+namespace GraphQL.Federation.Resolvers;
+
+/// <summary>
+/// Resolves the <c>_entities</c> field for GraphQL Federation.
+/// </summary>
+public sealed class EntityResolver : IFieldResolver
+{
+    /// <inheritdoc/>
+    private EntityResolver()
+    {
+    }
+
+    /// <summary>
+    /// Returns the static instance of <see cref="EntityResolver"/>.
+    /// </summary>
+    public static EntityResolver Instance { get; } = new EntityResolver();
+
+    /// <summary>
+    /// Converts representations to a list of <see cref="Representation"/> objects.
+    /// This should occur during field validation so that the representations can be validated.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">
+    /// Occurs when the requested type cannot be found, is not an object graph type, has not been
+    /// configured for GraphQL Federation, or cannot be converted to the source type.
+    /// </exception>
+    /// <remarks>
+    /// Exceptions thrown within this method are expected to be returned to the caller as a validation error
+    /// (aka Input Error), not logged as a server error (aka Processing Error).
+    /// </remarks>
+    public IEnumerable<Representation> ConvertRepresentations(ISchema schema, IList representations)
```

Comment:
I need to read this implementation carefully to understand it. Unfortunately, I can hardly find time for it...

Reply:
Understandable.  That's why I'm breaking apart #3921 into segments.  So we can review small pieces at a time and get through it all with good review for everything, even if it takes a few weeks to do so.  Similarly, it is easier to realize what parts of the new code needs better documentation or tests.